Change installation guide (#1510)
Deltares/hydromt: docs/overview/installation.rst

Reviewer: I think it might be good to add a section where we explain the main difference between pixi/uv and conda, e.g. directory based vs global env activation.
And perhaps that and how uv/pixi environments can be activated like a conda env

Installation guide
==================

HydroMT is available on PyPI and conda-forge.
The most common usage of HydroMT is through a model plugin, e.g. via hydromt_wflow, hydromt_sfincs etc.
These plugins install HydroMT as a dependency.
However, if you want to use HydroMT directly, e.g. to develop your own model plugin or to use the HydroMT Python API without a plugin,
you can install HydroMT using the following steps.

.. _installation_prerequisites:

Prerequisite: python installation
=================================

You'll need **Python 3.11 or greater** and a package manager such as uv, pixi, or others in order to use HydroMT.
These package managers help you to install (Python) packages and

> `manage environments <https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html>`_

Reviewer: should we update this link?

such that different installations do not conflict.

If you do not yet have such a package manager, we recommend using either:

- `uv <https://docs.astral.sh/uv/>`_: uses `pypi.org <https://pypi.org>`_ for downloading dependencies.

> - `pixi <https://pixi.sh>`_: uses `conda-forge <https://conda-forge.org/>`_ for downloading dependencies.

Reviewer: pixi does both

.. _installation_hydromt:

Installing HydroMT
==================

HydroMT is available from PyPI and conda-forge, and can be installed using uv, pixi, or others.
Here we will describe the installation using **uv** and the **pixi** package manager.

Basic installation
------------------

We strongly recommend installing HydroMT in a separate environment to avoid conflicts with other packages.

> Therefore we use uv or pixi, which installs Python directly in the project folder.

Reviewer 2: Might be useful to make it clearer that we are creating a project folder with these instructions rather than just a new environment? 

```suggestion
Therefore we use uv or pixi to create a new project folder with Python directly installed. 
```

To install these optional dependencies, you can use the following uv/pip commands:

.. tab-set::
  :sync-group: package-manager

  .. tab-item:: uv
    :sync: uv

    .. code-block:: console

      $ uv pip install "hydromt[io]"
      $ uv pip install "hydromt[extra]"
      $ uv pip install "hydromt[examples]"
      $ uv pip install "hydromt[slim]"

  .. tab-item:: pixi
    :sync: pixi

>     Not required, as pixi install via conda-forge, which contains all optional dependencies by default.

Author: That's incorrect copilot
